python-checks/src/main/java/org/sonar/python/checks/CommentedCodeCheck.java

```diff
@@ -130,26 +121,25 @@ private static boolean isEmpty(String text) {
   private static boolean isTextParsedAsCode(String text) {
     try {
       AstNode astNode = parser.parse(text);
-      List<AstNode> expressions = astNode.getDescendants(PythonGrammar.EXPRESSION_STMT);
-      return astNode.getNumberOfChildren() > 1 && !isSimpleExpression(expressions);
+      FileInput parse = new PythonTreeMaker().fileInput(astNode);
+      return parse.statements() != null && !isSimpleExpression(parse);
     } catch (Exception e) {
       return false;
     }
   }
 
-  private static boolean isSimpleExpression(List<AstNode> expressionStatements) {
-    if (expressionStatements.size() != 1) {
+  private static boolean isSimpleExpression(FileInput fileInput) {
+    if (fileInput.statements().statements().size() > 1) {
       return false;
     }
-    AstNode expressionStatement = expressionStatements.get(0);
-    return (expressionStatement.getNumberOfChildren() == 1 && expressionStatement.getFirstChild().is(PythonGrammar.TESTLIST_STAR_EXPR))
-      || expressionStatement.hasDirectChildren(PythonGrammar.ANNASSIGN);
+    Statement statement = fileInput.statements().statements().get(0);
+    return statement.is(Tree.Kind.EXPRESSION_STMT) || statement.is(Tree.Kind.ANNOTATED_ASSIGNMENT);
   }
 
   private static List<List<Trivia>> groupTrivias(Token token) {
     List<List<Trivia>> result = new LinkedList<>();
```

Comment:
I know this is not from you, but can you change to use arrayList, there is no good usage of linked list.